### ISSUE TITLE
Unneeded WARNING

### ISSUE THREAD

**Issue by LeffeG** [zonemaster/zonemaster#1210]
When i do a test i don't want to get a WARNING when i have NS servers on different IP-net bit on same IP prefix especially if i have NS servers on more than one prefix. Some of the addresses are also anycast addresses.

WARNING  The following name server(s) are announced in the same IPv6 prefix (2001:XXX::/32): "ns1.XX.net/2001:XXX:100::2;ns2.XX.net/2001:XXX:200::2....
WARNING  The following name server(s) are announced in the same IPv6 prefix (2a02:YYY::/29): "ns1.YY.se/2001:YYY:10::1;ns2.YY.net/2001:YYY:10::2....

Is there a way to suppress these annoying WARNINGS?

**Comment by matsduf** [comment]
Thank you for reporting this. I think your issue is about the same as #1174. We will take a look at the logic of [CONNECTIVITY04].

If you have your own installation of Zonemaster you can create a copy of profile.json and then adjust the level of the message tag and use that for your testing.

Anycasting is a different dimension that Zonemaster cannot see. For ASN and prefixes we use a public database. For anycast we would need a public database of anycasted addresses. We do not know of any.

[CONNECTIVITY04]: https://github.com/zonemaster/zonemaster/blob/master/docs/public/specifications/tests/Connectivity-TP/connectivity04.md

**Comment by LeffeG** [comment]
Well, prefix and subnet is not the same.
Prefix is taken from the LIR WHOIS and subnet can be something else below the announced prefix.
I made this change and it looks good to me.
```
--- /usr/share/perl5/auto/share/dist/Zonemaster-Engine/profile.json.org 2023-09-12 10:11:11.000000000 +0200
+++ /usr/share/perl5/auto/share/dist/Zonemaster-Engine/profile.json     2023-09-28 00:08:10.952158098 +0200
@@ -101,9 +101,9 @@
             "CN04_EMPTY_PREFIX_SET": "ERROR",
             "CN04_ERROR_PREFIX_DATABASE": "ERROR",
             "CN04_IPV4_DIFFERENT_PREFIX": "INFO",
-            "CN04_IPV4_SAME_PREFIX": "WARNING",
+            "CN04_IPV4_SAME_PREFIX": "INFO",
             "CN04_IPV6_DIFFERENT_PREFIX": "INFO",
-            "CN04_IPV6_SAME_PREFIX": "WARNING",
+            "CN04_IPV6_SAME_PREFIX": "INFO",
             "EMPTY_ASN_SET" : "ERROR",
             "ERROR_ASN_DATABASE" : "ERROR",
             "IPV4_DIFFERENT_ASN" : "INFO",
```

**Comment by matsduf** [comment]
Just be aware that profile.json in default path will be overwritten at update.

The prefixes used by Zonemaster are the prefixes found in routing, not in the LIR table.

**Comment by LeffeG** [comment]
Yes understood, thanks for the advice.
Understood, but I would like to think that a /32 IPv6 prefix is routed below the announced BGP AS and I do have NS servers on 2 different AS'es.
Would it be sufficient to lower this WARNING to INFO if at least 1 NS exist on 2 different AS per protocol (IPv4 or IPv6)?

**Comment by matsduf** [comment]
>  Would it be sufficient to lower this WARNING to INFO if at least 1 NS exist on 2 different AS per protocol (IPv4 or IPv6)?

Yes, I think that WARNING should be only if all NS, per protocol, are on the same subnet. See #1174. Maybe a NOTICE if you have more than one NS on a subnet. I plan to create a proposed update soon.

The meaning of the different levels (INFO, NOTICE, WARNING) can be found in https://github.com/zonemaster/zonemaster/blob/master/docs/public/specifications/tests/SeverityLevelDefinitions.md
